src/gluonts/mx/trainer/learning_rate_scheduler.py

```diff
@@ -11,21 +11,55 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
+from dataclasses import field
 from typing import Dict, Any
 
+from pydantic.dataclasses import dataclass
+from typing_extensions import Literal
+
 import numpy as np
 import mxnet as mx
 from mxnet import gluon
 import mxnet.gluon.nn as nn
 
-from gluonts.core.component import validated, logger
-from gluonts.exceptions import GluonTSUserError
+from gluonts.core.component import validated
 
 from .callback import Callback
 
 
-class MetricAttentiveScheduler(mx.lr_scheduler.LRScheduler):
-    r"""
+@dataclass
+class Metric:
```

Comment:
Is `Metric` the right name? Seems like the sub-classes do not define actual metrics, but only a strategy for what value to keep track of. Maybe `Objective` is a better term?